simple_certmanager/admin_views.py

```diff
@@ -0,0 +1,145 @@
+from django import forms
+from django.core.files.base import ContentFile
+from django.views import generic
+
+from OpenSSL import crypto
+
+from simple_certmanager.constants import CertificateTypes
+from simple_certmanager.models import Certificate, CertificateSigningRequestInfo
+
+
+class CertificateSigningRequestInfoForm(forms.ModelForm):
+    class Meta:
+        model = CertificateSigningRequestInfo
+        fields = [
+            "country_name",
+            "organization_name",
+            "state_or_province_name",
+            "email_address",
+            "common_name",
+        ]
+        widgets = {
+            "country_name": forms.TextInput(attrs={"class": "form-control"}),
+            "organization_name": forms.TextInput(attrs={"class": "form-control"}),
+            "state_or_province_name": forms.TextInput(attrs={"class": "form-control"}),
+            "email_address": forms.EmailInput(attrs={"class": "form-control"}),
+            "common_name": forms.TextInput(attrs={"class": "form-control"}),
+        }
+
+
+class GenerateCertificateView(generic.edit.CreateView):
+    template_name_suffix = "_form"
+    model = CertificateSigningRequestInfo
+    form_class = CertificateSigningRequestInfoForm
+    success_url = "/admin/simple_certmanager/certificate/"
```

Comment:
this will break as soon as someone puts the admin under a different URL in their project